### PR TITLE
Added make backup

### PR DESCRIPTION
**Pull request by igorkorsukov**

### CODE REVIEW

File: src/project/internal/notationproject.cpp [hunk 1]
```diff
@@ -337,6 +337,31 @@ mu::Ret NotationProject::doSave(bool generateBackup)
     return make_ret(Ret::Code::Ok);
 }
 
+mu::Ret NotationProject::makeCurrentFileAsBackup()
+{
+    io::path filePath = m_engravingProject->path();
+    if (io::suffix(filePath) != engraving::MSCZ) {
+        LOGW() << "backup alowed only for MSCZ, current: " << filePath;
```
Comment (Jojo-Schmitz) [inline src/project/internal/notationproject.cpp:344]: Typo: `allowed`. And probably `currently`

File: src/project/internal/notationproject.cpp [hunk 2]
```diff
@@ -337,6 +337,31 @@ mu::Ret NotationProject::doSave(bool generateBackup)
...
+
+    if (!fileSystem()->exists(filePath)) {
+        LOGD() << "file not exists, probably not saved yet";
+        return make_ret(Ret::Code::Ok);
+    }
+
+    io::path backupFilePath = filePath + "~";
+    Ret ret = fileSystem()->move(filePath, backupFilePath, true);
+    if (!ret) {
+        LOGE() << "failed move from: " << filePath << ", to: " << backupFilePath;
```
Comment (Jojo-Schmitz) [inline src/project/internal/notationproject.cpp:356]: `to move` or `moving`

File: src/project/internal/notationproject.cpp [hunk 3]
```diff
@@ -337,6 +337,31 @@ mu::Ret NotationProject::doSave(bool generateBackup)
...
+mu::Ret NotationProject::makeCurrentFileAsBackup()
+{
+    io::path filePath = m_engravingProject->path();
+    if (io::suffix(filePath) != engraving::MSCZ) {
+        LOGW() << "backup alowed only for MSCZ, current: " << filePath;
+        return make_ret(Ret::Code::Ok);
+    }
+
+    if (!fileSystem()->exists(filePath)) {
+        LOGD() << "file not exists, probably not saved yet";
```
Comment (Jojo-Schmitz) [inline src/project/internal/notationproject.cpp:349]: `does not exist`

File: src/framework/system/systemerrors.h [hunk 4]
```diff
@@ -55,6 +56,7 @@ inline Ret make_ret(Err e)
     case Err::FSWriteError: return Ret(retCode, trc("system", "An error occurred when writing to the file"));
     case Err::FSMakingError: return Ret(retCode, trc("system", "An error occurred when making a path"));
     case Err::FSCopyError: return Ret(retCode, trc("system", "An error occurred when coping the file"));
+    case Err::FSMoveErrors: return Ret(retCode, trc("system", "An error occurred when move the file or dir"));
```
Comment (Jojo-Schmitz) [inline src/framework/system/systemerrors.h:59]: `moving`

File: src/project/internal/notationproject.cpp (hunk 1 above)
Comment (igorkorsukov) [reply src/project/internal/notationproject.cpp]: Thanks, fixed!

File: src/project/internal/notationproject.cpp (hunk 2 above)
Comment (igorkorsukov) [reply src/project/internal/notationproject.cpp]: Thanks, fixed!

File: src/project/internal/notationproject.cpp (hunk 3 above)
Comment (igorkorsukov) [reply src/project/internal/notationproject.cpp]: Thanks, fixed!

File: src/framework/system/systemerrors.h (hunk 4 above)
Comment (igorkorsukov) [reply src/framework/system/systemerrors.h]: Thanks, fixed!

**Review by RomanPudashkin — APPROVED** [review]

**Pull request merged**